### PR TITLE
Add support for options handling in log and stashes views #1661

### LINKED ISSUE #1661: Ignore whitespace does not work in Log [2] view
**Describe the bug**
Hi, truly an amazing project, and thank you for making it. I can click `.` for more options, type `o` for options, and then navigate to "Ignore whitespace" and change to true.

This works (!) thank you. But it only works for "Status [1]" view. But it does not work for "Log [2]" view after navigating to a commit.

**To Reproduce**
1. click `.` for more options
2. type `o` for options
3. navigate to "Ignore whitespace"
4. Change it to true
5. click `tab` to open Log [2]
6. choose commit with up/down arrows
7. press enter
8. choose file with up/down arrows
9. when seeing file commit diff on the right whitespaces are still visible

**Expected behavior**
No whitespaces in dif.

**Screenshots**
image

**Context (please complete the following information):**
 - OS/Distro + Version: Fedora29
 - GitUI Version 0.22.1
 - Rust version: no rust on the machine

**Additional context**
I tried following https://github.com/extrawurst/gitui/blob/master/src/tabs/revlog.rs to find missing link of options. I see at https://github.com/extrawurst/gitui/blob/289eddd9beeffeb57ca3ae9f55380dd05996c23a/asyncgit/src/sync/commit_files.rs#L24 the options are not passed, but I do not know about the codebase.

I posted first on https://github.com/extrawurst/gitui/issues/1147 but then created new issue.

### PR DESCRIPTION
**Pull request by kamillo**
This Pull Request fixes/closes #1661.

It changes the following:
- Adds support for options handling in log and stashes views
- `SharedOptions` are now passed to `InspectCommitComponent` and `CompareCommitsComponent` and used instead of `DiffOptions::default()` 

I followed the checklist:
- [ ] I added unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [x] I added an appropriate item to the changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * crashes on entering submodules ([#1510](https://github.com/extrawurst/gitui/issues/1510))
 * fix race issue: revlog messages sometimes appear empty ([#1473](https://github.com/extrawurst/gitui/issues/1473))
 * default to tick-based updates [[@cruessler](https://github.com/cruessler)] ([#1444](https://github.com/extrawurst/gitui/issues/1444))
+* add support for options handling in log and stashes views ([#1661](https://github.com/extrawurst/gitui/issues/1661))
```
Comment (extrawurst) [inline CHANGELOG.md:53]: Please annotate with your GitHub handle (see other entries)

**Review by extrawurst — CHANGES_REQUESTED** [review]
Looks good. Thanks for taking this! One small comment

File: CHANGELOG.md (hunk 1 above)
Comment (kamillo) [reply CHANGELOG.md]: Sure, done.

**Comment by extrawurst** [comment]
@pm100 mind giving this a spin? I am currently on holiday and can’t test it

**Comment by pm100** [comment]
Not quite sure how this has happened (I am also away but have my macbook and can remote into my home windows dev system) something has pulled crossterm 0.26 into gitui. This has a serious breaking change for windows. I already ran across this for tiu-textarea - see https://github.com/rhysd/tui-textarea/pull/17 because , if you recall, I am working on integrating that into gitui (I have pending PR too, hopefully they get to that)

I supposed that its not this PR but something earlier BUT my initial testing suggests its this change (I pulled gitui master branch and it was fine)

The fix is simple if you know where the keyboard event conversion core is. You now need to explictly test that the crossterm key event is a keydown (ct 0.26 on windows also generates a key release event - grr)

If you tell me where to tweak that I can edit it in and test it (remote debugging / exploring of code I dont know is very hard - only one monitor, small screen, and v slow)

BTW. This PR works fine on Mac and Ubuntu (via WSL).

**Comment by kamillo** [comment]
Let me know if you need any further changes for this PR.

**Comment by pm100** [comment]
@kamillo I was testing this on windows and hit a blocking issue pulled in from crossterm 0.26, this is now fixed in master and you have synced with that. I will finish testing tomorrow or the next day

**Comment by pm100** [comment]
@extrawaust tested on macos 13.3.1 ubuntu 20.2 (WSL) and windows 11. Works fine

**Pull request merged**